Tests/Functional/Action/Platform/OidcLoginAuthenticationActionTest.php

```diff
@@ -0,0 +1,187 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Bundle\Lti1p3Bundle\Tests\Functional\Action\Platform;
+
+use Lcobucci\JWT\Parser;
+use OAT\Library\Lti1p3Core\Launch\Builder\OidcLaunchRequestBuilder;
+use OAT\Library\Lti1p3Core\Launch\Request\OidcLaunchRequest;
+use OAT\Library\Lti1p3Core\Link\ResourceLink\ResourceLink;
+use OAT\Library\Lti1p3Core\Message\LtiMessage;
+use OAT\Library\Lti1p3Core\Message\LtiMessageInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\Library\Lti1p3Core\Security\Jwt\AssociativeDecoder;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class OidcLoginAuthenticationActionTest extends WebTestCase
+{
+    /** @var KernelBrowser */
+    private $client;
+
+    /** @var RegistrationInterface */
+    private $registration;
+
+    protected function setUp(): void
+    {
+        $this->client = static::createClient();
+
+        $this->registration = static::$container
+            ->get(RegistrationRepositoryInterface::class)
+            ->find('testRegistration');
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->client->getCrawler()->clear();
+    }
+
+    public function testValidLoginAuthenticationWithPostMethod(): void
+    {
+        /** @var OidcLaunchRequest $oidcLaunchRequest */
+        $oidcLaunchRequest = static::$container->get(OidcLaunchRequestBuilder::class)->buildResourceLinkOidcLaunchRequest(
+            new ResourceLink('resourceLinkIdentifier'),
+            $this->registration,
+            'loginHint'
+        );
+
+        $this->client->request(
+            Request::METHOD_POST,
+            '/oidc/login-authentication',
+            [
+                'scope' => 'openid',
+                'response_type' => 'id_token',
+                'client_id' => $this->registration->getClientId(),
+                'redirect_uri' => $this->registration->getTool()->getLaunchUrl(),
+                'login_hint' => 'loginHint',
+                'state' => 'state',
+                'response_mode' => 'form_post',
+                'nonce' => 'nonce',
+                'prompt' => 'none',
+                'lti_message_hint' => $oidcLaunchRequest->getLtiMessageHint(),
+                'lti_deploymentId' => $oidcLaunchRequest->getLtiDeploymentId(),
+            ]
+        );
+
+        $response = $this->client->getResponse();
```

Comment:
do you want me guys to use a dataprovider ?

Reply:
the delta between the 2 tests are basically GET vs POST.
let me know :)

Reply:
refactored, for Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php as well